sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java

```diff
@@ -146,4 +155,20 @@ private Mono<Response<AnalyzeSentimentResultCollection>> getAnalyzedSentimentRes
             .map(Utility::toAnalyzeSentimentResultCollectionResponse)
             .onErrorMap(Utility::mapToHttpResponseExceptionIfExists);
     }
+
+    private void throwIfCallingNotAvailableFeatureInAnalyzeSentimentOptions(AnalyzeSentimentOptions options) {
+        if (options == null) {
+            return;
+        }
+        if (options.isIncludeOpinionMining()) {
+            throwIfTargetServiceVersionFound(this.serviceVersion, Arrays.asList(TextAnalyticsServiceVersion.V3_0),
+                getUnsupportedServiceApiVersionMessage("AnalyzeSentimentOptions.includeOpinionMining",
+                    serviceVersion, TextAnalyticsServiceVersion.V3_1));
+        }
+        if (options.isServiceLogsDisabled()) {
+            throwIfTargetServiceVersionFound(this.serviceVersion, Arrays.asList(TextAnalyticsServiceVersion.V3_0),
+                getUnsupportedServiceApiVersionMessage("TextAnalyticsRequestOptions.disableServiceLogs",
+                    serviceVersion, TextAnalyticsServiceVersion.V3_1));
+        }
```

Comment:
should `throwIfCallingNotAvailableFeatureInOptions` be used here?

Reply:
Actually this method should be renamed to the private method `throwIfCallingNotAvailableFeatureInOptions` in `AnalyzeSentimentAysncClient`. The reason to keep them separate independent methods in each xxxAsycnClient class is to support future new options properties.